src/NServiceBus.Core/Unicast/RunningEndpointInstance.cs

```diff
@@ -61,32 +63,64 @@ public async Task Stop()
 
         public Task Send(object message, SendOptions options)
         {
+            Guard.AgainstNull(nameof(message), message);
+            Guard.AgainstNull(nameof(options), options);
+
+            CheckIfInRunnungsState();
             return messageSession.Send(message, options);
         }
 
         public Task Send<T>(Action<T> messageConstructor, SendOptions options)
         {
+            Guard.AgainstNull(nameof(messageConstructor), messageConstructor);
+            Guard.AgainstNull(nameof(options), options);
+
+            CheckIfInRunnungsState();
             return messageSession.Send(messageConstructor, options);
         }
 
         public Task Publish(object message, PublishOptions options)
         {
+            Guard.AgainstNull(nameof(message), message);
+            Guard.AgainstNull(nameof(options), options);
+
+            CheckIfInRunnungsState();
             return messageSession.Publish(message, options);
         }
 
         public Task Publish<T>(Action<T> messageConstructor, PublishOptions publishOptions)
         {
+            Guard.AgainstNull(nameof(messageConstructor), messageConstructor);
+            Guard.AgainstNull(nameof(publishOptions), publishOptions);
+
+            CheckIfInRunnungsState();
             return messageSession.Publish(messageConstructor, publishOptions);
         }
 
-        public Task Subscribe(Type eventType, SubscribeOptions options)
+        public Task Subscribe(Type eventType, SubscribeOptions subscribeOptions)
+        {
+            Guard.AgainstNull(nameof(eventType), eventType);
+            Guard.AgainstNull(nameof(subscribeOptions), subscribeOptions);
+            
+            CheckIfInRunnungsState();
+            return messageSession.Subscribe(eventType, subscribeOptions);
+        }
+
+        public Task Unsubscribe(Type eventType, UnsubscribeOptions unsubscribeOptions)
         {
-            return messageSession.Subscribe(eventType, options);
+            Guard.AgainstNull(nameof(eventType), eventType);
+            Guard.AgainstNull(nameof(unsubscribeOptions), unsubscribeOptions);
+
+            CheckIfInRunnungsState();
+            return messageSession.Unsubscribe(eventType, unsubscribeOptions);
         }
 
-        public Task Unsubscribe(Type eventType, UnsubscribeOptions options)
+        void CheckIfInRunnungsState()
         {
-            return messageSession.Unsubscribe(eventType, options);
+            if (stopping)
+            {
+                throw new InvalidOperationException("The endpoint instance is not in the running state.");
```

Comment:
Would "Invoking messaging operations on the endpoint instance after it has been stopped is not supported" better guide the users? Is there anything we can tell them to do in order to address the issue?